Network/NetworkService.swift

```diff
@@ -63,4 +72,34 @@ class NetworkService {
     func cancel() {
         task?.cancel()
     }
+    
+    
+    //MARK: Private
+    private func makeQuery(for url: URL, params: [String: Any], type: QueryType) -> URLRequest? {
+        switch type {
+        case .json:
+            var mutableRequest = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData,
+                                                     timeoutInterval: 10.0)
+            mutableRequest.httpBody = try! JSONSerialization.data(withJSONObject: params, options: [])
+            
+            return mutableRequest
+        case .path:
+            var query = ""
+            
+            for (key, value) in params {
+                query = query + key + "=" + (value as! String) + "&"
```

Comment:
I think here should be `+ ("\(value)") +` because the value might be Bool - like in my real project that I am implementing this `QueryType` thing.